src/crab/type_domain.hpp

```
@@ -3,131 +3,144 @@
-    TypeDomain operator|(const TypeDomain& other) const { return TypeDomain{inv | other.inv}; }
-    TypeDomain operator|(TypeDomain&& other) const { return TypeDomain{inv | std::move(other.inv)}; }
-
-    std::optional<TypeDomain> meet(const TypeDomain& other) const {
-        if (auto res = this->inv & other.inv) {
-            return TypeDomain{std::move(res)};
-        }
-        return {};
-    }
-    bool operator<=(const TypeDomain& other) const { return inv <= other.inv; }
-    void set_to_top() { inv.set_to_top(); }
+    std::optional<TypeDomain> meet(const TypeDomain& other) const;
```

Clang (14.0.6)

[warning] function 'meet' should be marked [[nodiscard]]

(modernize-use-nodiscard)